src/services/worker/tasks/burn-rates.task.ts

```diff
@@ -26,294 +11,150 @@ export class BurnRatesTask {
-              record.cluster,
-            ),
-          async (result: any) => {
-            const balance = result.includes('InsufficientFunds') ? 0 : +result;
-            if (Number.isNaN(balance)) {
-              throw new Error('Failed to fetch balance');
-            }
-            record.balance = balance;
-          },
-          async (result: AtomicRetryFailedResult) => {
-            if (
-              result.solidityError &&
-              this._solidityErrorsService.isError(
-                result.solidityError,
-                'ClusterIsLiquidated',
-              )
-            ) {
-              await this._clusterService.update(
-                {
-                  owner: record.owner,
-                  operatorIds: record.operatorIds,
-                },
-                {
-                  isLiquidated: true,
-                },
-              );
-              console.error(
-                `Marked cluster as liquidated and skipping it: ${JSON.stringify(
-                  { ...record, isLiquidated: false },
-                )}`,
-                `. Reason: ${JSON.stringify({
-                  result,
-                  record,
-                })}`,
-              );
-              return;
-            }
-            throw new Error(
-              `Web3Provider.getBalance failed: ${JSON.stringify({
-                result,
-                record,
-              })}`,
-            );
-          },
-        );
-
-        // Cluster liquidated state
-        await this._atomicRetry(
-          async () =>
-            Web3Provider.isLiquidated(
-              record.owner,
-              record.operatorIds,
-              record.cluster,
-            ),
-          async (result: boolean) => {
-            if (typeof result !== 'boolean') {
-              throw new Error('Failed to fetch liquidated status');
-            }
-            record.isLiquidated = result;
-          },
-          async (result: AtomicRetryFailedResult) => {
-            throw new Error(
-              `Web3Provider.isLiquidated failed: ${JSON.stringify({
-                result,
-                record,
-              })}`,
-            );
-          },
-        );
-
-        // Current block number
-        let currentBlockNumber: number = null;
-        await this._atomicRetry(
-          async () => Web3Provider.currentBlockNumber(),
-          async (result: any) => {
-            if (!Number.isInteger(result)) {
-              throw new Error('Failed to fetch current block number');
-            }
-            currentBlockNumber = result;
+    const missedRecords = (
+      await this._clusterService.findBy({
+        where: { burnRate: null },
+      })
+    ).map(item => {
+      try {
+        item.cluster = JSON.parse(item.cluster);
+      } catch (e) {
+        console.error(`Can not json parse cluster data from DB! Skipping.`);
+      }
+      return item;
+    });
+
+    const burnRates = await Promise.allSettled(
```

Comment:
Instead of requesting burnRate for all 100 clusters, and then balance for 100 clusters etc, - It should ask for all the parts for one cluster at a time.

```
try:
    oneClusterData = await Promise.all([getBurnRate, getBalance, isLiquidated, ...]);
    // One cluster data is fully ready - process it
catch:
    // retry or whatever
```

`Promise.all` is by purpose, because If any part of cluster data is not loaded - it should fail and continue with next cluster (or with the same, you can decide)

Because with current approach if you have many clusters - it will reach limit rates for user who uses alchemyapi (or other services apis with rate limits), and will never be synced. I had this issue with current implementation and free alchemyapi.

Also if you want to increase capacity of parallel clusters resolution you can do this:

// pseudocode
```
cluster1 = clustersList.pop();
...
clusterN = clustersList.pop();

// ! -> Promise.allSettled(Promise.all, Promise.all, ...)
parallelClustersData = await Promise.allSettled([
   Promise.all([cluster1.getBurnRate, cluster1.getBalance, cluster1.isLiquidated, ...]),
   ...
   Promise.all([clusterN.getBurnRate, clusterN.getBalance, clusterN.isLiquidated, ...]),
])

// process chunk of clusters here
```

`N` can be by default 5-10, and should be able to read it from arguments or env var as all the rest variables.
In this case if user uses api with rate limits and network have many clusters, user can adjust amount of parallel clusters to resolve to his maximum and it will be fair and proper solution.

Please refactor this place again.